faunadb-scala/src/main/scala/faunadb/FaunaClient.scala

```diff
@@ -363,8 +370,12 @@ class FaunaClient private (connection: Connection) {
   }
 
   private def handleNetworkExceptions[A]: PartialFunction[Throwable, Future[A]] = {
-    case ex: ConnectException => Future.failed(new UnavailableException(ex.getMessage, ex))
-    case ex: TimeoutException => Future.failed(new TimeoutException(ex.getMessage))
+    case ex: ConnectException =>
+      Future.failed(new UnavailableException(ex.getMessage, ex))
+    case ex: TimeoutException =>
+      Future.failed(new TimeoutException(ex.getMessage))
+    case ex: CompletionException if ex.getMessage.contains("too many concurrent streams") =>
```

Comment:
this is actually a `CompletionException` wrapping the `IOException` with `too many concurrent streams`

Reply:
Same comment about checking on the cause type here as well:

```scala:
case ex: CompletionException if ex.getCause.isInstanceOf[IOException] && ex.getMessage.contains("too many concurrent streams")
```

:)